File: Smallest Non-Zero Number.java
// Problem: Smallest Non-Zero Number
// geeksforgeeks problem of the day -> 14th July 2026
// JAVA CODE
class Solution {
    public int find(int[] arr) {
      long need = 0; 
      for(int i=arr.length-1;i>=0;i--){
          need = (need + arr[i] + 1)/2;
      }
      return (int)need;
    }
}

File: Smallest Non-Zero Number.cpp
// Problem: Smallest Non-Zero Number
// geeksforgeeks problem of the day -> 14th July 2026
// C++ CODE
class Solution {
  public:   
    int find(vector<int>& arr) {
       int curr = 0;
       for(int i = arr.size()-1 ; i>=0 ; i--){
           curr+= arr[i];
           curr = (curr+1)/2;
       }
       return curr;
    }
};